js/apps/admin-ui/src/user/user-credentials/ResetPasswordDialog.tsx

```diff
@@ -157,7 +161,19 @@ export const ResetPasswordDialog = ({
             <PasswordInput
               data-testid="passwordField"
               id="password"
-              {...register("password", { required: true })}
+              onChange={(e) => {
+                onChange(e);
+                if (
+                  passwordConfirmation !== (e.target as HTMLInputElement).value
```

Comment:
I believe you can use [`currentTarget`](https://developer.mozilla.org/en-US/docs/Web/API/Event/currentTarget) instead, which should always be stable reference. This also means that you should not need to add a type-cast.
```suggestion
                  passwordConfirmation !== e.currentTarget.value
```